### PR TITLE
[Layout foundations] Rebuild Page Header with layout components

### LINKED ISSUE #7916: [Layout foundations] Rebuild `Page Header` with layout primitives
This component does a lot of content shifting when elements exist or don't exist via grid template areas. We may require a component that allows grid-area names to be passes, or at least a conversation on how we can handle complex, responsive layouts

### PR DESCRIPTION
**Pull request by kyledurand**
Closes https://github.com/Shopify/polaris/issues/7916

Spin link: https://admin.web.web-tdr8.kyle-durand.us.spin.dev/store/shop1

**Tophatting links**
[Details page](https://5d559397bae39100201eedc1-tmrswumsiu.chromatic.com/?path=/story/playground--details-page)
[Page stories](https://5d559397bae39100201eedc1-tmrswumsiu.chromatic.com/?path=/story/all-components-page--default)

### CODE REVIEW

File: polaris-react/src/components/Page/components/Header/Header.scss [hunk 1]
```diff
@@ -86,43 +36,9 @@ $action-menu-rollup-computed-width: 40px;
...
-    // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
-    align-items: center;
-  }
 }
 
 .TitleActionMenuWrapper {
-  // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
-  flex: 1 1 auto;
-
   .Header:not(.mobileView) & {
```
Comment (aveline) [inline polaris-react/src/components/Page/components/Header/Header.scss]: Can this be removed? Seems like it's handled by `Box` now?

**Review by aveline — APPROVED** [review]
Nice! 🎉

File: polaris-react/src/components/Page/components/Header/Header.scss (hunk 1 above)
Comment (kyledurand) [reply polaris-react/src/components/Page/components/Header/Header.scss]: Nice catch! https://github.com/Shopify/polaris/pull/7929/commits/bc28da39c58cc8498e75b335fa90e99e70e6f3e6

Comment (chazdean) [reply polaris-react/src/components/Page/components/Header/Header.scss]: You might able able to remove the entire `TitleActionMenuWrapper` selector as I don't think its being used?

**Review by chazdean — APPROVED** [review]

**Pull request merged**